irispy/sji.py

```diff
@@ -1,20 +1,720 @@
+        lines: list
+            A list of `matplotlib.axvline` objects that have been plotted.
+
+        Notes
+        -----
+        keyword arguments are passed onto matplotlib.pyplot.plot
+        """
+
+        if not axes:
+            axes = wcsaxes_compat.gca_wcs(self.wcs)
+
+        axes.axvline(self.meta['SLTPX1IX'])
+
+
+class SJICube(object):
+    """
+    SJICube
+
+    A series of SJI Images
+
+    Parameters
+    ----------
+
+    Attributes
+    ----------
+
+    Examples
+    --------
+    >>> from irispy.sji import SJICube
+    >>> from irispy.data import sample
+    >>> sji = SJICube(sample.SJI_CUBE_1400)   # doctest: +SKIP
+
+    """
+    #pylint: disable=W0613,E1101
+    def __init__(self, input):
+        """Creates a new instance"""
+        if isinstance(input, str):
+            fits = pyfits.open(input, memmap=True, do_not_scale_image_data=True)
+            # TODO find a new masking value for unscaled data
+            #self.data = np.ma.masked_less_equal(fits[0].data, 0)
+            self.data = fits[0].data
+            reference_header = deepcopy(fits[0].header)
+            table_header = deepcopy(fits[1].header)
+            # fix reference header
+            if reference_header.get('lvl_num') == 2:
+                reference_header['wavelnth'] = reference_header.get('twave1')
+                reference_header['detector'] = reference_header.get('instrume')
+                reference_header['waveunit'] = "Angstrom"
+                reference_header['obsrvtry'] = reference_header.get('telescop')
+            # check consistency
+            if reference_header['NAXIS3'] != self.data.shape[0]:
+                raise ValueError("Something is not right with this file!")
+
+            number_of_images = self.data.shape[0]
+            metas = []
+            dts = fits[1].data[:, fits[1].header['TIME']]
+            self.slit_position_x = u.Quantity(fits[1].data[:, fits[1].header['SLTPX1IX']], 'arcsec')
+            self.slit_position_y = u.Quantity(fits[1].data[:, fits[1].header['SLTPX2IX']], 'arcsec')
+
+            # append info in second hdu to each header
+            for i in range(number_of_images):
+                metas.append(deepcopy(reference_header))
+                metas[i]['DATE_OBS'] = str(parse_time(reference_header['STARTOBS']) + timedelta(seconds=dts[i]))
+                # copy over the individual header fields
+                for item in fits[1].header[7:]:
+                    metas[i][item] = fits[1].data[i, fits[1].header[item]]
+                    if item.count('EXPTIMES'):
+                        metas[i]['EXPTIME'] = fits[1].data[i, fits[1].header[item]]
+
+            self._meta = metas
+        elif len(input) > 1:
+            self.data = input[0]
+            self._meta = input[1]
+            number_of_images = self.data.shape[0]
+
+        norm = []
+        cmap = []
+        for i in range(number_of_images):
+            norm.append(deepcopy(self[i].plot_settings['norm']))
```

Comment:
Got the following error when initializing using a single SJI file string:
```
AttributeError                            Traceback (most recent call last)
<ipython-input-14-db6437a43c2d> in <module>()
----> 1 sjicube = SJICube(sji1330_file)

~/sunpy_dev/ehsteve_irispy/irispy/sji.py in __init__(self, input)
    182         cmap = []
    183         for i in range(number_of_images):
--> 184             norm.append(deepcopy(self[i].plot_settings['norm']))
    185             cmap.append(deepcopy(self[i].plot_settings['cmap']))
    186 

~/sunpy_dev/ehsteve_irispy/irispy/sji.py in __getitem__(self, key)
    196         work.  Otherwise, a mapcube is returned."""
    197         if isinstance(key, int):
--> 198             return self._get_map(key)
    199         elif isinstance(key, slice):
    200             return SJICube((self.data[key, :, :], self._meta[key]))

~/sunpy_dev/ehsteve_irispy/irispy/sji.py in _get_map(self, index)
    189 
    190     def _get_map(self, index):
--> 191         return SJIMap(self.data[index, :, :], self._meta[index], mask=self.data[index, :, :].mask)
    192 
    193     def __getitem__(self, key):

AttributeError: 'numpy.ndarray' object has no attribute 'mask'
```